File: JavaAlgos/DataStructures/src/data/structures/Queue.java
package data.structures;

import java.util.ArrayList;

public class Queue {
	private ArrayList<Object> data = new ArrayList<Object>();
	
	public void enqueue(Object val) {
		this.data.add(val);
	}
	
	public Object dequeue() {
		return this.data.remove(0);
	}
	
	public void printQueue() {
		String str = "Queue => ";
		for (Object val : this.data) {
			str += " | " + val;
		}
		str += " <- First In & Last Out";
		System.out.println(str);
	}
}

File: data_structures/queue.js
class Queue {
  constructor() {
    this.data = [];
  }

  enQueue(val) {
    this.data.unshift(val);
  }

  deQueue() {
    return this.data.pop();
  }

  printQueue() {
    console.log(this.data);
  }
}

const q = new Queue();
q.enQueue(11);
q.enQueue(2);
q.enQueue(10);
q.enQueue(5);
q.printQueue();
q.deQueue();
q.printQueue();
